services/grpc/grpc.js

```diff
@@ -1,278 +1,211 @@
 import EventEmitter from 'events'
-import StateMachine from 'javascript-state-machine'
+import intersection from 'lodash.intersection'
 import { proxyValue } from 'comlinkjs'
-import snakecase from 'lodash.snakecase'
 import { status } from '@grpc/grpc-js'
-import validateHost from '@zap/utils/validateHost'
+import LndGrpc from 'lnd-grpc'
 import { grpcLog } from '@zap/utils/log'
-import WalletUnlocker from './walletUnlocker'
-import Lightning from './lightning'
+import lightningMethods from './lightning.methods'
+import lightningSubscriptions from './lightning.subscriptions'
 
-const WALLET_STATE_LOCKED = 'WALLET_STATE_LOCKED'
-const WALLET_STATE_ACTIVE = 'WALLET_STATE_ACTIVE'
+const GRPC_WALLET_UNLOCKER_SERVICE_ACTIVE = 'GRPC_WALLET_UNLOCKER_SERVICE_ACTIVE'
+const GRPC_LIGHTNING_SERVICE_ACTIVE = 'GRPC_LIGHTNING_SERVICE_ACTIVE'
 
 /**
- * Wrapper and controller for multiple GrpcService instances.
+ * LND gRPC wrapper.
  * @extends EventEmitter
  */
-class Grpc extends EventEmitter {
-  constructor() {
-    super()
-    this.fsm = new StateMachine({
-      init: 'pending',
-      transitions: [
-        { name: 'init', from: 'pending', to: 'ready' },
-        { name: 'activateWalletUnlocker', from: ['ready', 'active'], to: 'locked' },
-        { name: 'activateLightning', from: ['ready', 'locked'], to: 'active' },
-        { name: 'disconnect', from: ['locked', 'active'], to: 'ready' },
-      ],
-      methods: {
-        onBeforeActivateWalletUnlocker: this.onBeforeActivateWalletUnlocker.bind(this),
-        onBeforeActivateLightning: this.onBeforeActivateLightning.bind(this),
-        onBeforeDisconnect: this.onBeforeDisconnect.bind(this),
-        onAfterDisconnect: this.onAfterDisconnect.bind(this),
-        onLeaveActive: this.onLeaveActive.bind(this),
-      },
-    })
-    this.supportedServices = [WalletUnlocker, Lightning]
-    this.services = {}
-    this.fsm.init()
-  }
+class ZapGrpc extends EventEmitter {
+  init(options) {
```

Comment:
The reason for the `init` method is that we create a singleton instance of the `ZapGrpc` class, and we don't necessarily have the connection details on hand at the time when that first gets created (since it gets created when the web worker is first initialised)